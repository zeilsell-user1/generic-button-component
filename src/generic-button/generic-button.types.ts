//'enum like' object declaration (notice the `as const` at the end)
// used becasue exporting const enum will not work when imported
// into another module.

export const ButtonStyles = {
  ACTIVECTA: "ACTIVECTA",
  INACTIVECTA: "INACTIVECTA",
  BACK: "BACK",
  OKAY: "OKAY",
  CANCEL: "CANCEL",
} as const;

//'enum like' "TS type" creation
export type ButtonStylesType = keyof typeof ButtonStyles;
