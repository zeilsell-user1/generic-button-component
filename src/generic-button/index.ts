export { GenericButton } from "./generic-button";
export { ButtonTypes } from "./generic-button.types";
