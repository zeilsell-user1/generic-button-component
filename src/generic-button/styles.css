.button {
  cursor: pointer;
  border-radius: 5%;
  background: white;
  color: black;
  border: 1px solid black;
  margin: 1%;
}
