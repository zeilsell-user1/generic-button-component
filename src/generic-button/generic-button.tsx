import "./styles.css";

import type { ReactNode } from "react";
import React, { useEffect, useState } from "react";

import { ButtonTypes } from "./generic-button.types";

interface Props {
  children?: ReactNode;
  style: ButtonTypes;
  onCLick?: () => void;
  text?: string;
}

export const GenericButton = (props: Props): JSX.Element => {
  interface ButtonStyle {
    textColor: string;
    hoverColor: string;
    background: string;
    border: string;
    font: string;
  }

  const [buttonStyle, setButtonStyle] = useState({} as ButtonStyle);

  useEffect(() => {
    const styleDetails: ButtonStyle = {} as ButtonStyle;

    console.log("props = ",props);

    switch (props.style) {
      case ButtonTypes.ActiveCta: {
        styleDetails.background = "#e69122";
        styleDetails.textColor = "#30220f";
        styleDetails.hoverColor = "black";
        styleDetails.border = "#30220f";
        break;
      }
      case ButtonTypes.InactiveCta: {
        styleDetails.background = "#e3ddd5";
        styleDetails.textColor = "#ebcea9";
        styleDetails.hoverColor = "#ebcea9";
        styleDetails.border = "#24190a";
        break;
      }
      case ButtonTypes.Back: {
        styleDetails.background = "#f5f3f05";
        styleDetails.textColor = "#30220f";
        styleDetails.hoverColor = "#30220f";
        styleDetails.border = "#24190a";
        break;
      }
      case ButtonTypes.Okay: {
        styleDetails.background = "#f5f3f05";
        styleDetails.textColor = "#30220f";
        styleDetails.hoverColor = "#f76402";
        styleDetails.border = "#24190a";
        break;
      }
      case ButtonTypes.Cancel: {
        styleDetails.background = "#f5f3f05";
        styleDetails.textColor = "#30220f";
        styleDetails.hoverColor = "#30220f";
        styleDetails.border = "#24190a";
        break;
      }
      default:
        styleDetails.background = "#f5f3f05";
        styleDetails.textColor = "#30220f";
        styleDetails.hoverColor = "#30220f";
        styleDetails.border = "#24190a";
    }

    setButtonStyle(styleDetails);
  }, []);

  return (
    <button
      className={"button"}
      style={{
        background: buttonStyle.background,
        color: buttonStyle.textColor,
        //hoverColor:buttonStyle.hoverColor,
        //border:1px solid buttonStyle.border,
        font: buttonStyle.font,
      }}
    >
      {props.text}
      {props.children}
    </button>
  );
};
