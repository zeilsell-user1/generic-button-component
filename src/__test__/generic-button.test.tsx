import { ButtonTypes, GenericButton } from "../generic-button";

describe("genericbutton component", () => {
  it("should render an active CTA button", () => {
    const containerActiveCta = (): JSX.Element => (
      <GenericButton style={ButtonTypes.InactiveCta}>text</GenericButton>
    );
    expect(containerActiveCta).toMatchSnapshot();
  });

  it("should render an inactive CTA button", () => {
    const containerInactiveCta = (): JSX.Element => (
      <GenericButton style={ButtonTypes.InactiveCta}>text</GenericButton>
    );
    expect(containerInactiveCta).toMatchSnapshot();
  });

  it("should render a back button", () => {
    const containerBack = (): JSX.Element => (
      <GenericButton style={ButtonTypes.InactiveCta}>Back</GenericButton>
    );
    expect(containerBack).toMatchSnapshot();
  });

  it("should render an OK button", () => {
    const containerOkay = (): JSX.Element => (
      <GenericButton style={ButtonTypes.InactiveCta}>OK</GenericButton>
    );
    expect(containerOkay).toMatchSnapshot();
  });

  it("should render a cancel button", () => {
    const containerCancel = (): JSX.Element => (
      <GenericButton style={ButtonTypes.InactiveCta}>Cancel</GenericButton>
    );
    expect(containerCancel).toMatchSnapshot();
  });
});
