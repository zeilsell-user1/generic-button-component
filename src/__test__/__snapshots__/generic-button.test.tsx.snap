// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genericbutton component should render a back button 1`] = `[Function]`;

exports[`genericbutton component should render a cancel button 1`] = `[Function]`;

exports[`genericbutton component should render an OK button 1`] = `[Function]`;

exports[`genericbutton component should render an active CTA button 1`] = `[Function]`;

exports[`genericbutton component should render an inactive CTA button 1`] = `[Function]`;
